### DISCUSSION TITLE
Using ATTACH DATABASE on SQLite for multiple database locations

### DISCUSSION THREAD

**Discussion by AfdhalAtiffTan** [pocketbase/pocketbase#4241] [Q&A] [answered]
I have two data sources, one is to be updated slowly (maybe once a day) and another is to be updated frequently (around 10 times a second).

To preserve the SSD/HDD/eMMC/SD storage lifespan, I am planning to put the "slow database" on the disk and the "fast database" in the RAM (in-memory database).

Is it possible to combine both databases using ATTACH DATABASE command with the embedded SQLite?

**Comment by ganigeorgiev** [comment]
While technically it is doable, I think you'll be better with using SQLite directly.

Also note that PocketBase currently doesn't allow operating with an in-memory SQLite db (at least not out of the box).

> To preserve the SSD/HDD/eMMC/SD storage lifespan...

I don't understand your concerns but you could just have regular backups of your data (_and/or maybe check Litestream/Marmot/etc. streaming replication tools_) and avoid the complexity of managing 2 data sources altogether.

**Reply by AfdhalAtiffTan** [reply]
Thanks Gani for your reply.

I like Pocketbase due to the built-in Realtime API and authentication features.

I ran SQLite on an SBC with an on-board eMMC similar to a Raspberry Pi and I've seen many of them returning back for repairs after years in the field. We find that the eMMC is stuck in read-only mode after prolonged daily writes.

Any advice for me to try to implement this feature myself?

**Reply by ganigeorgiev** [reply, marked as answer]
It really depends on your specific use case and how you plan to operate and sync the data between the 2 data sources, so I'm not sure what to recommend.
The code that takes care for initializing the SQLite db connection can be found in https://github.com/pocketbase/pocketbase/blob/master/core/db_cgo.go#L44 (and the `db_nocgo.go` for the pure Go version).
If your data can be loaded entirely in memory then one possible workaround without much code changes could be to try to mount the `pb_data/data.db`  in something like `tmpfs` (_the entire `pb_data` directory or maybe at some other location combined with symlinks?_) and then use Litestream or other replication tool to copy the data to a persistent storage. But I have no idea whether this will work, especially when in WAL mode where there are also sidecar `-wal` and `-shm` files that you'll have to take care for.

**Reply by AfdhalAtiffTan** [reply]
Thanks Gani, I will look into Litestream.
